pandas/core/frame.py

```diff
@@ -1909,43 +1914,77 @@ def to_dict(self, orient: str = "dict", into=dict):
             elif orient.startswith("i"):
                 orient = "index"
 
+        if not index and orient not in ["split", "tight"]:
```

Comment:
I am not sure this is desireable. The default config of index should not lead to a ValueError

Reply:
Hey phofl! Thanks for the feedback! I assume you mean the default value for the index parameter - the default value for the index parameter is True, so it wouldn't lead to a ValueError. This check was directly taken from the implementation of the index parameter in 'to_json'. It checks that index is only True if orient is 'split' or 'tight'. A value error would be raised only when index is set to False AND orient is not 'split' or 'tight'.